src/repository/url.go

```
@@ -0,0 +1,47 @@
+}
+
+func (r *urlRepository) RegisterUrl(url entity.Url) (*entity.Url, genericerror.GenericError) {
+	err := r.db.Create(&url).Error
+	if err != nil {
+		return nil, genericerror.NewInternalErrByErr(err)
+	}
+
+	return &url, nil
+}
+
+func (r *urlRepository) GetOriginalUrlByShortUrl(shortUrl string) (string, genericerror.GenericError) {
```

golangci-lint (1.64.8)

[warning] var-naming: method GetOriginalUrlByShortUrl should be GetOriginalURLByShortURL

(revive)